packages/babel-plugin-proposal-decorators/test/fixtures/2021-12-misc--to-es2015/initProto-existing-derived-constructor/exec.js

```diff
@@ -1,28 +1,173 @@
-let self, a, initCalled;
+{
+  let self, a, initCalled;
 
-function deco(_, context) {
-  context.addInitializer(() => {
-    initCalled = true;
-  })
-}
+  function deco(_, context) {
+    context.addInitializer(() => {
+      initCalled = true;
+    });
+  }
 
-class B {
-  constructor(s) {
-    a = s;
+  class B {
+    constructor(s) {
+      a = s;
+    }
   }
-}
 
-class A extends B {
-  constructor() {
-    let a = 2;
-    self = super(a);
+  class A extends B {
+    constructor() {
+      let a = 2;
+      self = super(a);
+    }
+
+    @deco
+    method() {}
   }
 
-  @deco
-  method() {}
+  let instance = new A();
+  expect(self).toBe(instance);
+  expect(a).toBe(2);
+  expect(initCalled).toBe(true);
 }
 
-let instance = new A();
-expect(self).toBe(instance);
-expect(a).toBe(2);
-expect(initCalled).toBe(true);
+{
+  const dec = (fn) => {
+    return function () {
+      return fn.call(this) + 100;
+    };
+  };
+
+  class B {
+    constructor(v) {
+      this.a = v;
+    }
+    method() {
+      return this.a;
+    }
+  }
+
+  {
+    ("super() nested within another constructor should not be transformed");
+    let log = [];
+    class A extends B {
+      constructor() {
+        log.push(super(3).method());
+        new (class Dummy extends B {
+          constructor() {
+            log.push(super(4).method());
+          }
+        })();
+      }
+
+      @dec
+      method() {
+        return this.a;
+      }
+    }
+
+    new A();
+
+    expect(log + "").toBe("103,4");
+  }
+
+  {
+    ("super() not in decorated derived constructor should not be transformed: computed key");
+    let log = [];
+    new (class Dummy extends B {
+      constructor() {
+        let key;
+        class A extends B {
+          constructor() {
+            log.push(super(6).method());
+          }
+
+          @dec
+          method() {
+            return this.a;
+          }
+
+          [((key = super(5).method()), log.push(key), key)];
+        }
+
+        new A();
+      }
+    })();
+
+    expect(log + "").toBe("5,106");
+  }
+
+  {
+    ("super() in decorator expression within decorated derived constructor should be transformed: decorator expression");
+    let log = [];
+    const noop = () => (fn) => fn;
+    new (class extends B {
+      constructor() {
+        class A extends B {
+          constructor() {
+            log.push(super(8).method());
+          }
+
+          @dec
+          method() {
+            return this.a;
+          }
+
+          @noop(log.push(super(7).method())) noop() {}
+        }
+
+        new A();
+      }
+    })();
+
+    expect(log + "").toBe("7,108");
+  }
+
+  {
+    ("super() within decorated derived constructor should be transformed: computed key");
```

Comment:
This test will fail if we skip the whole `ClassExpression`, ditto for `ClassDeclaration`.